File: sdks/java/connect-sdk/src/main/java/com/mobiscroll/connect/Provider.java
package com.mobiscroll.connect;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/** Calendar provider supported by the Mobiscroll Connect API. */
public enum Provider {
    GOOGLE("google"),
    MICROSOFT("microsoft"),
    APPLE("apple"),
    CALDAV("caldav");

    private final String wire;

    Provider(String wire) {
        this.wire = wire;
    }

    @JsonValue
    public String wireValue() {
        return wire;
    }

    @JsonCreator
    public static Provider fromWire(String value) {
        if (value == null) {
            return null;
        }
        for (Provider p : values()) {
            if (p.wire.equalsIgnoreCase(value)) {
                return p;
            }
        }
        throw new IllegalArgumentException("Unknown provider: " + value);
    }
}

File: sdks/go/provider.go
package mobiscroll

// Provider identifies a calendar provider. The string value matches the API
// wire form, so no custom JSON marshaler is required.
type Provider string

// Supported providers.
const (
	ProviderGoogle    Provider = "google"
	ProviderMicrosoft Provider = "microsoft"
	ProviderApple     Provider = "apple"
	ProviderCalDav    Provider = "caldav"
)
